docs/src/gui/qtvcp-development.adoc

```diff
@@ -1,161 +1,206 @@
+
+*QtVCP source* is located in `+src/emc/usr_intf/qtvcp+` folder of
+LinuxCNC source tree.
+
+=== QtVCP Startup
+
+When QtVCP first starts:
+
+. It must decide if this object is a screen or a panel.
+. It searches for and collects information about paths of required files
+  and useful folders.
+. It then:
+.. Builds the HAL component,
+.. Loads the window instance,
+.. Adds handler extensions,
+.. Installs an event filter.
+
++
+Now the window/widgets are instantiated, the HAL pins are built. +
+This also initiates the `+_init_hal()+` function of the widgets.
+. The `+initialized__()+` handler function is called
+. The `STATUS` library is forced to update.
+. HAL component is set ready at this point.
+. A variety of optional switch arguments are set, including calling a
+  POSTGUI HAL file (if a screen).
+. Terminate signals are trapped and QtVCP now polls for events.
+
+=== QtVCP Shutdown
+
+Finally when QtVCP is asked to shutdown:
+
+. It calls shutdown functions in the handler file,
+. `STATUS` monitoring is shut down
+. HAL component gets killed
+
+== Path Information
+
+When QtVCP loads it collects paths informations.
+
+This is available in the handler file's `+__init__()+` function
+as *`path`*:
+
+*`IMAGEDIR`*:: Path of builtin images
+*`SCREENDIR`*:: Path of builtin motion controller screens
+*`PANELDIR`*:: Path of builtin accessory panels
+*`WORKINGDIR`*:: Path of where QtVCP was launched from
+*`CONFIGPATH`*:: Path of the launched configuration
+*`BASEDIR`*:: general path, used to derive all paths
+*`BASENAME`*:: Generic name used to derive all paths
+*`LIBDIR`*:: Path of QtVCP's Python library
+*`HANDLER`*:: Path of handler file
+*`XML`*:: Path of .ui file
+*`DOMAIN`*:: Path of translation
+*`IS_SCREEN`*:: screen/panel switch
 
 == Idiosyncrasies
 
 These try to cover non-obvious situations.
 
-=== Error code  collecting
-
-Linuxcnc's error code collecting can only be read from one place.
-When read , it is 'consumed' - no other object can read it.
-In QtVCP screens, it is recommended to use the ScreenOptionss widget to
-set up error reading. They are then sent to other objects via STATUS
-signals.
-
-=== Jog rate
-
-Linuxcnc has no internal record of jog rate - you must specify it at the
-time of jogging.
-QtVCP uses the STATUS library to keep track of the latest linear and
-angular jog rate. It is always specified in machine units per minute, so must be
-converted when in non-machine units mode.
-So if your machine is imperial based but you are in metric mode
-changes to jog rate sent to ACTION functions must be converted to imperial.
-at the same time if the machine is metric based and you are in imperial mode,
- changes to jog rate must be sent to ACTION functions in metric units.
-Angular jog rates the units don't change in metric/imperial
-mode so you can send them to ACTION functions without conversion.
+=== Error Code Collecting
+
+*LinuxCNC's error code collecting can only be read from one place*.
```

Comment:
Why bold? A bit too much bold in this paragraph to highlight the parts that really should be bold.